elasticdl/python/master/optimizer_wrapper.py

```diff
@@ -199,199 +150,71 @@ def _init_thread_local(self):
     def apply_gradients(self, grads_and_vars):
         """Update variable values.
 
-        Arguments:
-            grads_and_vars: A list of (gradient, variable) pairs.
-
+        Args:
+            grads_and_vars: A list of (gradient, variable) pairs. If the
+                variable is from ElasticDL embedding layer, it should be a
+                ElasticDL `Tensor` object. Otherwise it is a TensorFlow
+                variable.
         """
         # TODO (#1255): Discuss whether `OptimizerWrapper` needs a lock after
         # implementing PS.
         self._init_thread_local()
 
-        grads_and_vars = list(grads_and_vars)
-
-        # split `grads_and_vars` according to whether it is from
-        # ElasticDL embedding layer
-        grads_and_vars_local = []
-        grads_and_vars_kv_store = []
+        grads_and_vars_new = []
         for grad, var in grads_and_vars:
             layer_name = _get_embedding_layer_name_from_var(var)
-            if layer_name:
-                grads_and_vars_kv_store.append((grad, layer_name))
+            if not layer_name:
+                grads_and_vars_new.append((grad, var))
             else:
-                grads_and_vars_local.append((grad, var))
```

Comment:
`_get_embedding_layer_name_from_var` is used only once. We could just write such logic directly:

```python
for grad, var in grads_and_vars:
    if isinstance(var, str):
        grads_and_vars_new.append(self._handle_embedding_grad(grad, layer_name))
    else:
        grads_and_vars_new.append((grad, var))   
```

Reply:
Done